src/engine/ruleset/decoders/wazuh-agent/syscollector/syscollector-dbsync.yml

```diff
@@ -0,0 +1,108 @@
+# hwinfo
+#  {"data":{"board_serial":"0","checksum":"f7605acc23c70cbb76290a45d956dbab27f5eebc","cpu_cores":2,"cpu_mhz":2371.0,"cpu_name":"AMD Ryzen 5 4500U with Radeon Graphics","ram_free":1691948,"ram_total":2024444,"ram_usage":17,"scan_time":"2023/03/10 03:37:16"},"operation":"MODIFIED","type":"dbsync_hwinfo"}
+# ports
+#  {"data":{"checksum":"631a5cbb816346c903708bbc14cbc50597272601","inode":26875,"item_id":"fffd4d8c20cacb413ae6c1f39a3a52389375701d","local_ip":"10.0.2.15","local_port":68,"pid":0,"process":null,"protocol":"udp","remote_ip":"10.0.2.2","remote_port":67,"rx_queue":0,"scan_time":"2023/03/14 12:41:19","state":null,"tx_queue":0},"operation":"INSERTED","type":"dbsync_ports"}
+# packages
+#  {"data":{"architecture":"x86_64","checksum":"0d7e29ef72b657c96512b43efa6637ad0431b2e5","description":"Tool to analyse BIOS DMI data","format":"rpm","groups":"Unspecified","install_time":"1607103514","item_id":"bbb009a26c1b5bc26c093e9fbcd9b01d0fd227d4","location":null,"multiarch":null,"name":"dmidecode","priority":null,"scan_time":"2023/03/14 15:09:43","size":222289,"source":null,"vendor":"CentOS","version":"1:3.2-6.el8"},"operation":"DELETED","type":"dbsync_packages"}
+# network_iface
+#  {"data":{"adapter":null,"checksum":"052a00a969cb2d54a97890b20ea9af156b74d42d","item_id":"d4aa8b01955e438235d586585492f3f5edceec1b","mac":"52:54:00:27:8b:50","mtu":1500,"name":"eth0","rx_bytes":510595,"rx_dropped":0,"rx_errors":0,"rx_packets":4522,"scan_time":"2023/03/10 03:27:12","state":"up","tx_bytes":616053,"tx_dropped":0,"tx_errors":0,"tx_packets":4186,"type":"ethernet"},"operation":"MODIFIED","type":"dbsync_network_iface"}
+# network_address
+#  {"data":{"address":"fe80::a00:27ff:fe0d:a1e5","broadcast":null,"checksum":"7bdb80f090fb4aa51756a4fd0382a1b08ee3e0be","dhcp":"unknown","iface":"eth1","item_id":"4a53b37f2fb0d3ef21c7ff8b5e1b9072b578b82f","metric":null,"netmask":"ffff:ffff:ffff:ffff::","proto":1,"scan_time":"2023/03/14 15:32:01"},"operation":"INSERTED","type":"dbsync_network_address"}
+# network_protocol
+# {"data":{"checksum":"ea17673e7422c0ab04c4f1f111a5828be8cd366a","dhcp":"unknown","gateway":"192.168.0.1|600","iface":"enp4s0","item_id":"9dff246584835755137820c975f034d089e90b6f","metric":" ","type":"ipv6"},"operation":"INSERTED","type":"dbsync_network_protocol"}
+# hotfixes
+# {"data":{"scan_time":"2021/10/29 14:26:24","hotfix":"KB123456","checksum":"abcdef0123456789"}, "type":"dbsync_hotfixes","operation":"MODIFIED"}
+
+
+
+normalize:
+  - check: +ef_exists/~json.data.address OR +ef_exists/~json.data.mac OR +ef_exists/~json.data.os_version OR +ef_exists/~json.data.release
+    map:
+      - ~do_update_kvdb: true
+
+  - check: +s_ne/~json.data.hostname/ OR +s_ne/~json.data.os_name/ OR +s_ne/~json.data.os_platform/ OR +s_ne/~json.data.sysname/
```

Comment:
Wouldn't it be better to use `+ef_exists/~json.data.hostname` instead of `+s_ne/~json.data.hostname/`?

Reply:
te helper not equal to empty will be true if this field contains any string and exist, the exist won't check if the value of that field is empty.